core-primitives/attestation-handler/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+#[macro_use]
+extern crate sgx_tstd as std;
+
+// re-export module to properly feature gate sgx and regular std environment
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+pub mod sgx_reexport_prelude {
+	pub use serde_json_sgx as serde_json;
+	pub use thiserror_sgx as thiserror;
+	pub use yasna_sgx as yasna;
+}
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
```

Comment:
Are we sure the attestation handler only works for `sgx`? We should always try to make it work in both the `std` AND `sgx` feature when possible.

Reply:
I had another look at this, but I think it would be quite an effort to compile it with std in a meaningful way. There are quite a few direct calls into sgx functions that we would need to isolate/replace:
```
use sgx_rand::{os, Rng};
use sgx_tcrypto::{rsgx_sha256_slice, SgxEccHandle};
use sgx_tse::{rsgx_create_report, rsgx_verify_report};
```
Of course I could write empty dummy functions for the std case but I'm not sure this makes sense as long as we have no use for them (no unit tests)

Reply:
Alright, that's reason enough for me to keep it in `sgx` only (for now) 👍